### ISSUE TITLE
Can we write `_sigmoid_cross_entropy_with_logits` in a simple way?

### ISSUE THREAD

**Issue by AIWanderer-X** [TencentARC/Open-MAGVIT2#21]
https://github.com/TencentARC/Open-MAGVIT2/blob/6bfab153fb2fe612edbf8afa9b3a607d8337c194/taming/modules/losses/vqperceptual.py#L33C1-L33C56

For the code:
```python
def _sigmoid_cross_entropy_with_logits(labels, logits):
    """
    non-saturating loss 
    """
    zeros = torch.zeros_like(logits, dtype=logits.dtype)
    condition = (logits >= zeros)
    relu_logits = torch.where(condition, logits, zeros)
    neg_abs_logits = torch.where(condition, -logits, logits)
    return relu_logits - logits * labels + torch.log1p(torch.exp(neg_abs_logits))
```

Can we just write `F.relu(logits) - logits * labels + torch.log1p(torch.exp(-torch.abs(logits)))`? Why does it have to be such complex?

**Comment by RobertLuo1** [comment]
Hi, We just follow the code in MAGVIT. I think your version should also work fine.
